logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/JoinConversationViaCodeUseCase.kt

```diff
@@ -0,0 +1,67 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationGroupRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.id.toModel
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.network.api.base.authenticated.conversation.ConversationMemberAddedResponse
+
+/**
+ * Use case for joining a conversation via a code invite code.
+ * the param can be obtained from the deep link
+ * @param code The code of the conversation to join.
+ * @param key The key of the conversation to join.
+ * @param domain optional domain of the conversation to join.
+ */
+class JoinConversationViaCodeUseCase internal constructor(
+    private val conversionsGroupRepository: ConversationGroupRepository,
+    private val selfUserId: UserId
+) {
+    suspend operator fun invoke(code: String, key: String, domain: String?): Result =
+        conversionsGroupRepository.joinViaInviteCode(code, key, null)
```

Comment:
can you explain the null in a comment , or maybe put as todo in case it's neccessary